uint8 data_len
DesiredPosition[] position
---
bool success